### DISCUSSION TITLE
Connection to local instance without network

### DISCUSSION THREAD

**Discussion by valrusu** [sijms/go-ora#381] [Q&A]
Hi,
Is it possible to connect to a local instance, which maybe does not have a listener running, similar to the below?
Thanks

```
echo $ORACLE_SID
DB1

echo $ORACLE_HOME
/export/home/oracle/app/product/12.2.0/dbhome_1

sqlplus / as sysdba
...
Connected to:
Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production


sqlplus myuser/password
...
Connected to:
Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production
```

**Comment by algosoup** [comment]
Could someone help please? I have the same question

**Comment by sijms** [comment]
you can use unix socket in case of linux machine or windows authentication in case of windows

**Reply by valrusu** [reply]
Thank you.
My understanding as of now is that OS authentication is working only on Windows (not my case) and Bequeath transport is not supported (usr/pwd to local instance, using ORACLE_SID and not through TCP). For these cases I use Godror, for the rest Go-ora. I just wrote a wrapper OpenDB function which first tries the go-ora driver, if that fails it tries the godror driver. Once the *sql.DB is obtain, I only work with it through the database/sql functions so it should be transparent as to which driver is behind. Hope this helps.

**Comment by sijms** [comment]
for os authentication windows is supported not linux
for linux you can use unix_socket and pass a url as follow
```golang
urlOptions := map[string]string {
    "SID": "YOUR SID",
    "UNIX SOCKET": "your socket file", // for example /usr/tmp/.oracle/sEXTPROC1
}
// no service name because you are using SID otherwise use service name
conn, err := sql.Open("127.0.0.1", 1521, "", user, password, urlOptions);
```

**Comment by valrusu** [comment]
Thank you again and thanks for this great package.

The socket would not work either in my case because the main purposes are to not have to provide a password and/or to connect when I do not have a listener running. The socket still needs it.
